tests/e2e/tests/critical-path/tree-view/tree-view-improvements.e2e.ts

```diff
@@ -0,0 +1,169 @@
+        await verifyKeysDisplayedInTheList([keynameSingle]);
+
+        await t.click(browserPage.setDeleteButton);
+        // verify that first element in the tree is "keys" and other folders are closed
+        await verifyKeysDisplayedInTheList([keynameSingle]);
+        await verifyKeysNotDisplayedInTheList([`${keyNames[0]}:1`, `${keyNames[2]}:2`]);
+
+        // switch between browser view and tree view
+        await t.click(browserPage.browserViewButton)
+            .click(browserPage.treeViewButton);
+        // Switch to tree view
+        await browserPage.deleteKeyByName(keyNames[4]);
+        await t.click(browserPage.clearFilterButton);
+        // get first folder name
+        const firstTreeItemText = await browserPage.getTextFromNthTreeElement(0);
+        const firstTreeItemKeys = Selector(`[data-testid="node-item_${firstTreeItemText}:keys:keys:"]`); // keys after node item opened
+        // verify that the first folder with namespaces is expanded and selected
+        await t.expect(firstTreeItemKeys.visible)
+            .ok('First folder is not expanded');
+        await verifyKeysDisplayedInTheList([`${firstTreeItemText}:1`, `${firstTreeItemText}:2`]); // verify created keys are visible
+
+        const commands1 = [
+            `HSET ${keyNames[4]} field value`
+        ];
+        // Create 4 keys and index
+        await cliPage.sendCommandsInCli(commands1);
+        await t.click(browserPage.refreshKeysButton); // refresh keys
+        await t.expect(firstTreeItemKeys.visible)
+            .ok('Folder is not selected');
+        await verifyKeysDisplayedInTheList([`${firstTreeItemText}:1`, `${firstTreeItemText}:2`]); // verify created keys are visible
+
+        await browserPage.selectFilterGroupType(KeyTypesTexts.Hash);
+        await t.expect(firstTreeItemKeys.visible).ok('Folder is not selected after searching with HASH');
+        await verifyKeysDisplayedInTheList([`${firstTreeItemText}:1`, `${firstTreeItemText}:2`]); // verify created keys are visible
+
+        await browserPage.searchByKeyName('*');
+        await t.expect(firstTreeItemKeys.visible).ok('Folder is not selected');
+        await verifyKeysDisplayedInTheList([`${firstTreeItemText}:1`, `${firstTreeItemText}:2`]); // verify created keys are visible
+
+        await t.click(browserPage.clearFilterButton);
+        await t.expect(firstTreeItemKeys.visible).ok('Folder is not selected');
+        await verifyKeysDisplayedInTheList([`${firstTreeItemText}:1`, `${firstTreeItemText}:2`]); // verify created keys are visible
+
+        await browserPage.selectFilterGroupType(KeyTypesTexts.Stream);
+        await t.expect(browserPage.keyListTable.textContent).contains('No results found.', 'Key is not found message not displayed');
+
+        await t.click(browserPage.streamDeleteButton); // clear stream from filter
+        await t.expect(
+            Selector('[role="rowgroup"]').find('div').withText('No results found.').visible)
```

Comment:
just use here
`await t.expect(browserPage.keyListTable.textContent).notContains('No results found.', 'Key is not found message still displayed');`